python-spec/src/somacore/coordinates.py

```diff
@@ -0,0 +1,460 @@
+        inv_a = np.linalg.inv(self._matrix[:-1, :-1])
+        b2 = -inv_a @ self._matrix[:-1, -1].reshape((self.output_rank, 1))
+        inv_augmented: npt.NDArray[np.float64] = np.vstack(
+            (
+                np.hstack((inv_a, b2)),
+                np.hstack((np.zeros(self.output_rank), np.array([1]))),
+            )
+        )
+        return AffineTransform(self.output_axes, self.input_axes, inv_augmented)
+
+
+class ScaleTransform(AffineTransform):
+    """A scale coordinate transformation from one coordinate space to another.
+
+    Args:
+        input_axes: The names of the axes for the input coordinate space.
+        output_axes: The names of the axes for the output coordinate space.
+        scale_factors: The scale factors for the transformation. May be a single
+            value for isotropic (uniform) scale transformations or a value per axis
+            for non-isotropic transformations.
+
+    """
+
+    def __init__(
+        self,
+        input_axes: Union[str, Sequence[str]],
+        output_axes: Union[str, Sequence[str]],
+        scale_factors: npt.ArrayLike,
+    ):
+        super(AffineTransform, self).__init__(input_axes, output_axes)
+        if self.input_rank != self.output_rank:
+            raise ValueError("Incompatible rank of input and output axes")
+
+        self._scale_factors: Union[np.float64, npt.NDArray[np.float64]] = np.array(
+            scale_factors, dtype=np.float64
+        )
+        if self._scale_factors.size == 1:
+            self._scale_factors = self._scale_factors.reshape((1,))[0]
+            self._isotropic = True
+        elif self._scale_factors.size == self.input_rank:
+            self._scale_factors = self._scale_factors.reshape((self.input_rank,))
+            if np.all(self._scale_factors == self._scale_factors[0]):
+                self._scale_factors = self._scale_factors[0]
+                self._isotropic = True
+            else:
+                self._isotropic = False
+        else:
+            raise ValueError(
+                f"Scale factors have unexpected shape={self._scale_factors.shape} "
+                f"for a transform with rank={self.input_rank}."
+            )
+
+    def __matmul__(self, other: Any) -> CoordinateTransform:
+        if isinstance(other, CoordinateTransform):
+            if self.input_axes != other.output_axes:
+                raise ValueError("Axis mismatch between transformations.")
+            if isinstance(other, ScaleTransform):  # Includes IdentityTransform
+                return ScaleTransform(
+                    other.input_axes,
+                    self.output_axes,
+                    self.scale_factors * other.scale_factors,
+                )
+            if isinstance(other, AffineTransform):
+                return AffineTransform(
+                    other.input_axes,
+                    self.output_axes,
+                    self.augmented_matrix @ other.augmented_matrix,
```

Comment:
Maybe a comment for the non-experts (we're all polyglots, and not all image experts) that `@` maps to the `__matmul__` dunder-method